https://nodejs.dev/error-handling-in-nodejs

## 定义错误 Error objects
```js
throw new Error('Ran out of coffee')
```

Handling exceptions
```js
try {
  //lines of code
} catch (e) {}
```

## Exceptions with promises
```js
doSomething1()
  .then(doSomething2())
  .then(doSomething3())
  .catch(err => console.error(err))
```

## Error handling with async/await
```js
async function someFunction() {
  try {
    await someOtherFunction()
  } catch (err) {
    console.error(err.message)
  }
}
```

## Catching uncaught exceptions
If an uncaught exception gets thrown during the execution of your program, your program will crash.

To solve this, you listen for the uncaughtException event on the process object:
```js
process.on('uncaughtException', err => {
  console.error('There was an uncaught error', err)
  process.exit(1) //mandatory (as per the Node.js docs)
})
```